### DISCUSSION TITLE
What Should Our Name Be?

### DISCUSSION THREAD

**Discussion by mcb2003** [odilia-app/odilia-prototype#5] [Announcements]
Today I woke up to [this community post from NCommander](https://www.youtube.com/channel/UCWyrVfwRL-2DOkzsqrbjo5Q/community?lb=Ugkx0Qn3gWgX-0CVNiW7Yco0c8V69k0xV3Tf) in my YouTube subscriptions feed, and realised we had a small issue:

> I love how Yggdrasil Plug-N-Play Linux has a 900 number for support...

"Hmm, another project with the name Yggdrasil", I thaught. You may think this isn't a big deal, but in truth, this isn't the first time I've come across another project called Yggdrasil:

```
$ paru -Ss yggdrasil
community/yggdrasil 0.4.0-1 [2.25MiB 9.57MiB]
    An experiment in scalable routing as an encrypted IPv6 overlay network
aur/popura-git 0.3.15+popura2-2 [+1 ~0.01]
    Popura ポプラ: alternative Yggdrasil network client
aur/yggdrasil-git 0.4.2.r1.g87e9361-1 [+1 ~0.04]
    An experiment in scalable routing as an encrypted IPv6 overlay network (develop branch)
aur/popura 0.4.2+popura1-1 [+1 ~0.01]
    Popura ポプラ: alternative Yggdrasil network client
aur/yggdrasil-openrc 20190305-1 [+0 ~0.00]
    None
aur/autoygg-git v0.2.2.dev20210316233552-1 [+0 ~0.00]
    An easy internet egress client/server for Yggdrasil network.
aur/syg_go 0.2.1-1 [+0 ~0.00]
    Yggdrasil address miner written in Go
aur/yggdrasil-sr 0.1b-1 [+0 ~0.00]
    Yggdrasil Screenreader -- Make Linux Accessible!
```

That's right, we're in the AUR now, thanks @TTWNO! But I digress.

[Yggdrasil](https://yggdrasil-network.github.io/) is a name-independent, end-to-end encrypted routing scheme. It's an open-source project, and they seem to have chosen their name for exactly the same reason as us (it [unites worlds](https://en.wikipedia.org/wiki/Yggdrasil)).

And Yggdrasil has some other issues too:

* It's hard to spell, which is even more of an issue in the blind community, because, for the most part, we don't see words written down, we hear them synthesised by text-to-speech engines. Most everyone I've talked to so far thinks it's spelled "igdrasil", and that could make Yggdrasil hard to find on search engines.
* It's hard to pronounce. As @brogar2000 said in [the Yggdrasil audiogames.net thread](https://forum.audiogames.net/topic/42170/yggdrasil-a-linux-screen-reader-initiative-first-prototype-is-out/), this will lead people to refer to Yggdrasil as "that Linux screen reader" or "that Linux screen reader that's not Orca" in conversation. That can't exactly be good for our brand recognition! Even in my small friend group, people struggle to pronounce it.
* It's long (9 letters to be precise) and quite hard to type. Once upon a time, IRC only allowed a maximum of 9 letters in your username!
* The link between a giant tree that unites worlds and a screen reader is *there* (screen readers unite blind people with their sighted counterparts), but it's a little tenuous. In my humble opinion, it makes more sense for something like a network protocol, which is no doubt why it's being used for that.
* We don't exactly have many [domain options](https://www.namecheap.com/domains/registration/results/?domain=yggdrasil). We could use yggdrasil-sr, but that's not exactly ideal, as it's harder to remember.

So, I proposed we change our name, to make our project easier to find, and easier to remember. Here's what we came up with!

## Blab

This suggestion is older than Yggdrasil itself, having been suggested to me by @brogar2000 when I told him that I'd love to start a project like this some day. Of course, then we *did* start a project like this, and we decided on @albertotirla's suggestion of Yggdrasil.

To [blab](https://www.merriam-webster.com/dictionary/blab) is to engage in "idle or excessive talk", which I'm sure most screen reader users can attest to happening quite often with screen readers. :p

Blab could handily also be a [bacronym](https://en.wikipedia.org/wiki/Backronym) for "Blind Linux Accessibility Bridge", or even a [recursive bacronym](https://en.wikipedia.org/wiki/Acronym#Recursive_acronyms) for "BLAB Linux Accessibility for the Blind"!

### Pros

* Short
* Easy to type and spell
* Easy to pronounce (it's a single syllable)
* Linked to speech, and thus screen readers, in an obvious way.

## Cons

* May sound childish to some people
* Used by a few small-ish projects ([1](https://github.com/tuwukee/blab), [2](https://github.com/aoh/blab)) but not anything huge
    * I'm not counting the [Bedford Lab](https://github.com/blab) because it's an organisation, not a project, but feel free to if you wish
* [Not very many domain options](https://www.namecheap.com/domains/registration/results/?domain=blab), so we'd have to go with blab-sr.

## Chatter

This is a last-minute suggestion, but while looking up the exact definition of blab, I came across Chatter. It [has a very similar meaning to blab](https://www.merriam-webster.com/dictionary/chatter): "to talk idly, incessantly, or fast"

### Pros

* Short
* Easy to spell
* Easy to pronounce (for English speakers at least)

### Cons

* Lots of existing projects, the largest of which appears to be [this](https://github.com/thedevdojo/chatter)
* [Almost no domain options](https://www.namecheap.com/domains/registration/results/?domain=chatter), so again we'd have to go with chatter-sr

## Odilia

[Odilia](https://en.wikipedia.org/wiki/Odile_of_Alsace) is the patron saint of good eye sight in the Catholic and Eastern Authodox Christian denominations. This one was suggested by @TTWNO.

### Pros

* Fairly easy to spell (it's not a strange spelling at least)
* Fairly easy to pronounce
* Linked to screen readers by eye sight
* Almost nothing on Github with exactly the same name
* [Lots of good domain options](https://www.namecheap.com/domains/registration/results/?domain=odilia)

### Cons

* Has ties to religions still actively being practiced, so some people may not like it

## Vocem (or similar)

[vocem](https://www.wordhippo.com/what-is/the/latin-word-for-47ef65890ddbd7da73531268916c1d5697dd13e5.html) is Latin for voice.

### Pros

* Short
* Easy to type and spell
* Easy to pronounce (but see below)
* [Pretty good domain options](https://www.namecheap.com/domains/registration/results/?domain=vocem) although there's no .org

### Cons

* Used by a few small-ish projects ([1](https://github.com/vdbwouter/Vocem), [2](https://github.com/andriuskv/vocem)) but not anything huge
* Espeak-ng pronounces it with a soft c, which I believe is incorrect, though do feel free to correct me in the comments on this discussion if I'm wrong

## Yggdrasil

Finally, we could always keep our old name. From the [Yggdrasil home page](https://yggdrasil-sr.github.io):

> In Norse mythology, Yggdrasil is a huge tree around which the nine worlds gravitate...

### Pros

* No name change required

### Cons

* Long
* Hard to type and spell
* Hard to pronounce
* Already taken by [Yggdrasil Networking](https://github.com/yggdrasil-networking), who have ~1.8k stars on their most popular repository, and who chose their name for the same reasons as us
* Also taken by the now discontinued [Yggdrasil Linux/GNU/X distribution](https://en.wikipedia.org/wiki/Yggdrasil_Linux/GNU/X), so searching for "Yggdrasil Linux" yields irrelevent results
* [Almost every domain is taken](https://www.namecheap.com/domains/registration/results/?domain=yggdrasil), including the Github org name, so we have to use yggdrasil-sr
* I'd argue that the link between Yggdrasil, a huge tree that unites worlds, and a screen reader, a project that unites blind computer users with their sighted counterparts, is more tenuous than, say, a word that directly connotes the concept of speech, or speaking, but that's subjective, so make your own decision here

## Something Else?

As with anything, picking a new name for Yggdrasil is not an easy task, and we'd love your help! If you have any suggestions, thaughts on the names above, or any other comments, feel free to discuss this topic in this thread. The more community input we can get on this, the better!

I've [created a NextCloud poll](https://nextcloud.thefakevip.xyz/apps/forms/oLqmprWxspMSWGKK) for you to vote on what should be the new name. Note however that we still reserve the right to make our own decision on what the name should be, although we *will* use this vote to gage the community's feelings on this matter.

I can't wait to here your feedback, let's all build a better Linux screen reader together!

**Comment by dangero2000** [comment]
My vote is with Odilia. It has a very unique meaning which I’ve never heard before.
Actually, it has 2 meanings. It means greater clarity for us because it reads things on the screen, so it gives us the vision we need. The second meaning is in the letters. The name is short, easy to spell, easy to pronounce, and most of all easy to remember.
It’s simple, and because it’s simple, it gives the user even more clarity. Though, perhaps it would be good to differentiate the name from the aforementioned religions by calling it Odilia SR.
Also, someone said on the audiogames.net topic about this that they're not sure why people are upset about long hard to pronounce names. I was going to comment in the topic itself, but I decided to withdraw from that increasingly toxic topic and share my opinions here where it matters most.
It's important and essential, as an open-source and anti-close-sourced advocate, that I try my best to make Linux a user-friendly platform as best as I can despite not being a developer. One of the steps towards that is to make sure that things are simpler. The reason the average user sticks to Windows and stays away from Linux is perceived difficulty. The  poster in question states that there are tuns of commands on Linux that are long and don't have the easiest spelling, and that's exactly why the screen reader name needs to be changed.
The average user will never touch Linux until we can simplify how it works. This means simple names for programs and controls, a better Graphical User Interface that allows you to accomplish everything you could think of without ever having to touch the terminal, and documentation that is not only easy to find, but easy to read and understand for the technically uninitiated. Keeping the name as Yggdrasil is only further cementing the mindset of Linux being convoluted and difficult for the average user, and we'll be destroying the chances for a world-wide change even more.
Let's start making a change. No more operating systems where companies have full control over how people can use their software. No more companies spying on their users through the OS. No more companies claiming they care about feedback while ignoring their users at the same time. Let's put control back in the hands of the average user, give them a system they can trust, and listen to and implement their suggestions when they come up.
Let's make a better operating system than Windows or Mac. It is possible, and the way we start is by making things simpler. How do we do that? By fixing the accessibility frameworks, making a screen reader that's powerful yet simpler to use like NVDA, and change the name of the screen reader to something simple as well. I now rest my case.

**Reply by TTWNO** [reply]
My vote is also for Odilia.

**Comment by trypsynth** [comment]
This pole seems broken. All votes still show at 0%

**Comment by TTWNO** [comment]
Please write responses here in Github. Apparently the straw poll is broken....

Man, if we can't even get the poll right, how can we be trusted to build this much more complex thing!? Lol!

**Comment by trypsynth** [comment]
I also vote for Idilia

**Comment by emperor-limitless** [comment]
odilia.

**Comment by juanreina921** [comment]
I vote for keeping the name yggdrasil, as that's what the screen reader originally stood for, and it has more meaning to it than just what the screen reader does. BTW the google problem is solved, write yggdrasil screen reader on google and the page has all about the screen reader.

**Comment by dangero2000** [comment]
"It has more meaning to it than just what the screen reader does." Yes you're right. So does Odilia. Odilia doesn't directly mean clarity, but it means good eye sight which could be a way of saying good sight, so clarity. Clarity, not just for the sake of reading things on the screen, but clarity in the sense that it's simple, and it's paving the way for a simpler operating experience.

**Comment by albertotirla** [comment]
personally, I voted for keeping the name as is because of the significance it has, the symbol it convays. Odilia, if thought of like that, can also be a good symbol going forward, so it won't be so bad if that winns. Btw, the straw pole is working again, there are three votes for keeping the original name, the rest is 0. DK if anyone voted for anything else on there, but it seemns to work again, so worth a try.

**Comment by mcb2003** [comment]
Ok, I don't understand this. Just cast my vote (For Odilia since I think @brogar2000 made an excellent case for it to be honest) and Strawpoll still doesn't show it? Granted I haven't used StrawPoll very much, but how hard could this possibly be? I've never seen this happen before.

**Comment by albertotirla** [comment]
OK, it's definitely weird. I made another straw poll afew minutes ago, got the same result of it not showing votes properly...if at all. So, I think we should use this thread, it's not going to get very crouded anyway.

**Comment by mcb2003** [comment]
I made a new [NextCloud poll](https://nextcloud.thefakevip.xyz/apps/forms/oLqmprWxspMSWGKK), so please vote there now. Sorry about all this!

**Comment by trypsynth** [comment]
OK, I think I voted? I can't see the results, so shrug.

**Reply by albertotirla** [reply]
indeed, that's the only way so far, at least we have a valid poll option now. When the voting period will end, we could publish the data upon request.

**Comment by trypsynth** [comment]
Out of curiosity, how long will the pole go for?

**Reply by albertotirla** [reply]
around one week should be enough for everyone to vote.
